### PR TITLE
Fill out an advanced Conda section.

### LINKED ISSUE #375: Conda advanced topics TODO
- [ ] Jinja templates inside of `meta.yaml`: https://conda.io/docs/building/meta-yaml.html#templating-with-jinja
- [ ] tests: in BioConda keep them simple, just check if installation was successful
- [ ] use tarballs: how to get tarballs from GitHub/Mercurial: https://bioconda.github.io/guidelines.html#stable-urls
- [ ] preprocessing selector: https://conda.io/docs/building/meta-yaml.html#preprocessing-selectors
- [ ] the entire concept of "You provide us with one recipe, but we build 8 different packages out of it" (e.g. 4 python packages for Linux, 4 for OS X)
- [ ] library pinning: how and why: https://bioconda.github.io/guidelines.html#c-c

### PR DESCRIPTION
**Pull request by jmchilton**
Fixes #375.

### CODE REVIEW

File: topics/dev/tutorials/conda/slides.html [hunk 1]
```diff
@@ -494,6 +494,28 @@
...
+
+```yaml
+requirements:
+  build:
+    - gcc   # [not osx]
+    - llvm  # [osx]
+    - zlib
+```
+
+```
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: Add `yaml`

File: topics/dev/tutorials/conda/slides.html [hunk 2]
```diff
@@ -670,3 +702,397 @@
...
+CONDA_R: "3.3.2"
+CONDA_PERL: "5.22.0"
+CONDA_NPY: "112"
+CONDA_NCURSES: "5.9"
+CONDA_GSL: "1.16"
+CONDA_GMP: "5.1"
+CONDA_HDF5: "1.8.17"
+MACOSX_DEPLOYMENT_TARGET: "10.9"
+```
+
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: Add footnote link https://github.com/bioconda/bioconda-recipes/blob/master/scripts/env_matrix.yml

**Comment by jmchilton** [comment]
Aren't you at a beach? Thanks for cleaning this up!

**Comment by bgruening** [comment]
Thanks John!

File: topics/dev/tutorials/conda/slides.html [hunk 3]
```diff
@@ -670,3 +702,400 @@
...
+.footnote[.center[https://github.com/bioconda/bioconda-recipes/blob/master/scripts/env_matrix.yml]]
+
+---
+
+class: left
+
+## Pinning Libraries - Use jinja
+
+> If your package links dynamically against a particular library, it is **often necessary to pin the version against which it was compiled, in order to avoid ABI incompatibilities**. Instead of hardcoding a particular version in the recipe, we use jinja templates to achieve this. This **helps ensure that all bioconda packages are binary-compatible with each other**. For example, bioconda provides an environnmnet variable `CONDA_BOOST` that contains the current major version of Boost.
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: `environment` is misspelled.

File: topics/dev/tutorials/conda/slides.html [hunk 4]
```diff
@@ -670,3 +702,400 @@
...
+
+https://bioconda.github.io/guidelines.html#c-c
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: Footnote style

File: topics/dev/tutorials/conda/slides.html [hunk 5]
```diff
@@ -670,3 +702,400 @@
...
+```yaml
+source:
+  fn: v0.7.15.tar.gz
+  url: https://github.com/lh3/bwa/archive/v0.7.15.tar.gz
+  md5: 54fdee953c5c256d36885a1c5c6b118c
+```
+
+> While supported by Conda, `git_url` and `git_rev` are not as stable as a git tarball. Ideally a github repo should have tagged releases that are accessible as tarballs from the “releases” section of the github repo.
+
+https://conda.io/docs/building/meta-yaml.html#preprocessing-selectors
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: Footnote style. Also link seems wrong.

File: topics/dev/tutorials/conda/slides.html [hunk 6]
```diff
@@ -670,3 +702,400 @@
...
+
+For PyPI packages
+
+```sh
+conda skeleton pypi <package_name>
+```
+
+- Builds likely correct `build.sh` and `meta.yaml`.
+- The test automatically added is probably sufficient for library, may need to write extra tests for command-line tools.
+- Recipes requiring `python` should build on Python 2.7, 3.4, and 3.5 by default, preprocessing selectors can be used with `build: skip` to skip targets.
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: Should be 2.7, 3.5 and 3.6.

File: topics/dev/tutorials/conda/slides.html [hunk 7]
```diff
@@ -670,3 +702,400 @@
...
+
+---
+
+class: left
+
+## simulate-travis
+
+Some example commands:
+
+<dl>
```
Comment (nsoranzo) [inline topics/dev/tutorials/conda/slides.html]: `--install-requirements` should probably be mentioned first.

**Comment by nsoranzo** [comment]
Thanks a bunch @jmchilton!

**Comment by nsoranzo** [comment]
Damn, too late! I was reviewing on the phone in the airport. @bgruening can you fix these? Or I can do it later.

**Comment by bgruening** [comment]
I have a bunch of fixes in a local branch and will add this as well.

**Pull request merged**